src/Mobs/Ocelot.cpp

```diff
@@ -0,0 +1,194 @@
+				}
+
+				MoveToPosition(a_Closest_Player->GetPosition());
+			}
+			else
+			{
+				if (IsBegging())
+				{
+					SetIsBegging(false);
+					m_World->BroadcastEntityMetadata(*this);
+				}
+			}
+		}
+	}
+
+	if (IsTame() && !IsSitting())
+	{
+		TickFollowPlayer();
+	}
+	else if (IsSitting())
+	{
+		StopMovingToPosition();
+	}
+
+	m_World->BroadcastEntityMetadata(*this);
+}
+
+
+
+
+
+void cOcelot::TickFollowPlayer()
+{
+	class cCallback :
+		public cPlayerListCallback
+	{
+		virtual bool Item(cPlayer * a_Player) override
+		{
+			OwnerPos = a_Player->GetPosition();
+			OwnerFlying = a_Player->IsFlying();
+			return true;
+		}
+	public:
+		Vector3d OwnerPos;
+		bool OwnerFlying;
+	} Callback;
+
+	if (m_World->DoWithPlayerByUUID(m_OwnerUUID, Callback))
+	{
+		// The player is present in the world, follow him:
+		double Distance = (Callback.OwnerPos - GetPosition()).Length();
+		if (Distance > 12)
+		{
+			if (!Callback.OwnerFlying)
+			{
+				Callback.OwnerPos.y = FindFirstNonAirBlockPosition(Callback.OwnerPos.x, Callback.OwnerPos.z);
+				TeleportToCoords(Callback.OwnerPos.x, Callback.OwnerPos.y, Callback.OwnerPos.z);
+			}
+		}
+		if (Distance < 2)
+		{
+			StopMovingToPosition();
+		}
+		else
+		{
+			if (!Callback.OwnerFlying)
+			{
+				MoveToPosition(Callback.OwnerPos);
+			}
+		}
+	}
+}
+
+
+
+
+
+void cOcelot::OnRightClicked(cPlayer & a_Player)
+{
+	if (!IsTame())
+	{
+		if (
+			IsBegging() &&
+			((a_Player.GetPosition() - GetPosition()).Length() <= 3)
+		)
+		{
+			cItems Items;
+			GetBreedingItems(Items);
+			if (Items.ContainsType(a_Player.GetEquippedItem().m_ItemType))
+			{
+				if (!a_Player.IsGameModeCreative())
+				{
+					a_Player.GetInventory().RemoveOneEquippedItem();
+				}
+
+				if (GetRandomProvider().RandBool(1.0 / 3.0))
+				{
+					// Taming succeeded
+					SetIsBegging(false);
+
+					SetMaxHealth(20);
+					SetIsTame(true);
+					SetOwner(a_Player.GetName(), a_Player.GetUUID());
+					SetOcelotType(GetRandomProvider().RandInt<int>(1, 3));
+					m_World->BroadcastEntityStatus(*this, esWolfTamed);
+					m_World->GetBroadcaster().BroadcastParticleEffect("heart", static_cast<Vector3f>(GetPosition()), Vector3f{}, 0, 5);
+				}
+				else
+				{
+					// Taming failed
+					m_World->BroadcastEntityStatus(*this, esWolfTaming);
+					m_World->GetBroadcaster().BroadcastParticleEffect("smoke", static_cast<Vector3f>(GetPosition()), Vector3f{}, 0, 5);
+				}
+			}
+		}
+		else
+		{
+			super::OnRightClicked(a_Player);
+		}
+	}
+	else if (a_Player.GetUUID() == m_OwnerUUID)
+	{
+		super::OnRightClicked(a_Player);
+		SetIsSitting(!IsSitting());
+	}
+	m_World->BroadcastEntityMetadata(*this);
+}
+
+
+
+
+
+void cOcelot::SpawnOn(cClientHandle & a_ClientHandle)
+{
+	super::SpawnOn(a_ClientHandle);
+	if (GetRandomProvider().RandBool(1.0 / 7.0))
+	{
+		m_World->SpawnMob(GetPosX(), GetPosY(), GetPosZ(), m_MobType, true);
+		m_World->SpawnMob(GetPosX(), GetPosY(), GetPosZ(), m_MobType, true);
```

Comment:
I don't know anywhere else in the code that it would run on every ocelot spawn.
And the infinite loop can be avoided by checking if the current ocelot isn't baby.